hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/utils/TestWriteBase.java

```diff
@@ -524,9 +528,7 @@ private void checkInstantState(HoodieInstant.State state, String instantStr) {
     }
 
     protected String lastCompleteInstant() {
-      return OptionsResolver.isMorTable(conf)
-          ? TestUtils.getLastDeltaCompleteInstant(basePath)
-          : TestUtils.getLastCompleteInstant(basePath, HoodieTimeline.COMMIT_ACTION);
+      return this.ckpMetadata.lastCompleteInstant();
```

Comment:
Why we must fetch the instant from ckp metadata?

Reply:
Updated.
Add a special branch to handle lockless multiple writers.
For multiple writes, the writer started first might finish later, similarly, the writer started later might finish first. So the last completed instant from ckp metadata might not be equals to the last completed instant from timeline.